lib/mumukit/flow/adaptive_item.rb

```diff
@@ -0,0 +1,57 @@
+module Mumukit::Flow
+  module AdaptiveItem
+    include Mumukit::Flow::Node
+    include Mumukit::Flow::Suggesting
+
+    required :assignment
+    required :children
+    required :parent
+    required :tags
+
+    delegate :passed?, :submitter, to: :@assignment
+
+    def has?(tag)
+      tags.include?(tag)
+    end
+
+    private
+
+    def set_adaptive_assignment!(submitter)
```

Comment:
I don't get the semantics of this method. What does `set_adaptive_assignment` mean?

Reply:
The name is bad because I didn't notice at the time that its visibility could be limited through the `private` keyword, and I didn't want anyone using the method outside of `mumukit-flow`, so I tried to make _very_ clear and put the word `adaptive` into it. It's bad, but...

The method does more than just skipping the assignment - that's why I'm not sure about `skip_assignment_if_pending!`. It's a prerequisite for being able to suggest the next exercise. I thought about 'setting' the assignment to an exercise because it's not something that occurs in our domain model (exercises don't have assignments), but we _do_ need it here in order to know what exercise comes next, because the assignment knows if it was easy, if it was hard, who submitted it, and more. It does have to skip it if it's pending, true, but it's secondary. It's probably the consequence of a questionable design decision, as in the current state I need this initialization for it to work, but I can't make it clear through an `initialize` method because `AdaptiveItem` isn't a class.

I'll think about this a little more.

Reply:
We decided to merge this PR in its current state, and implement the necessary changes to fix this in an upcoming PR.

Details in https://github.com/mumuki/mumxukit-flow/issues/9